### DISCUSSION TITLE
NumSideQPs and NumElemQPs

### DISCUSSION THREAD

**Discussion by Flolaffel** [idaholab/moose#21987] [Q&A Modules: General] [answered]
Hello,

I was trying to experiment with the Quadrature-Block a bit and wanted to check the resulting number of QPs with `NumSideQPs` and `NumElemQPs` but it seems like my application can't find the objects. The error I get is

```
*** ERROR ***
A 'NumSideQPs' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.
```

I guess it's because the .C files are located in the moose/test folder rather than in some module folder. But I don't quite know how to use the `MOOSE_LIBRARY_PATH environment variable` to resolve that issue. What do I have to do to access said objects?

**Comment by GiudGiud** [comment, marked as answer]
I think they are test objects. You ll have to pass the allow-test-objects flag to use them

I think you ll need to add the BUILD_TEST_OBJECTS boolean in the makefile too.
and possibly just move the source from moose/test to your repo in last resort

**Reply by Flolaffel** [reply]
1. The MOOSE [documentation](https://mooseframework.inl.gov/application_usage/command_line_usage.html) only shows the flag `--disallow-test-objects`. Doesn't that imply that `--allow-test-objects` is the default?
2. I tried `BUILD_TEST_OBJECTS := yes` in my Makefile but it didn't help. I also tried `BUILD_TEST_OBJECTS_LIB := yes` because that command is used in the moose_test-opt Makefile. Didn't help either.

I also copied the files in my repo but had no success with that either. Which is kind of confusing.

What else can I try?

**Reply by GiudGiud** [reply]
move the test source files in your app then recompile

**Reply by Flolaffel** [reply]
That's what I meant when I said

`I also copied the files in my repo but had no success with that either. Which is kind of confusing.` (and recompiled my app afterwards)

Still not working.

**Reply by GiudGiud** [reply]
Re-register them to your application. Not mooseTestApp. See the source file at the top

**Reply by Flolaffel** [reply]
I can't quite follow. You mean copy the .C into intro/src and the .h into intro/include and then recompile with `make`, right? Because that's what I did and it doesn't work.

**Reply by GiudGiud** [reply]
modify them too
open the source you ll see the call to register to MooseTestApp

**Reply by Flolaffel** [reply]
Ok well that worked. Thanks!

In case anyone with equally slim knowledge of C++ stumbles across this in the future:

I copied the files in question to my own app. Then I had to change `registerMooseObject("MooseTestApp", NumSideQPs);` to `registerMooseObject("introTestApp", NumSideQPs);` and run my input with the `--allow-test-objects` flag. Modifications to the Makefile are not necessary for this workaround to work.

**Reply by GiudGiud** [reply]
if enough users need these objects we can move them outside of the test objects and into the official objects. Please let us know, future readers of this thread
